matsim/src/main/java/org/matsim/core/network/NetworkUtils.java

```diff
@@ -1018,8 +1019,30 @@ public static boolean addDisallowedNextLinks(Link link, String mode, List<Id<Lin
 		DisallowedNextLinks disallowedNextLinks = getOrCreateDisallowedNextLinks(link);
 		return disallowedNextLinks.addDisallowedLinkSequence(mode, linkIds);
 	}
-	
+
 	public static void removeDisallowedNextLinks(Link link) {
 		link.getAttributes().removeAttribute(DISALLOWED_NEXT_LINKS_ATTRIBUTE);
 	}
+
+	public static void addAllowedMode(Link link, String mode) {
+		Set<String> modes = new HashSet<>(link.getAllowedModes());
+		modes.add(mode);
+		link.setAllowedModes(modes);
+	}
+
+	public static void removeAllowedMode(Link link, String mode) {
+		Set<String> modes = new HashSet<>(link.getAllowedModes());
+		modes.remove(mode);
+		link.setAllowedModes(modes);
+	}
+
+	/**
+	 * Removes the given modes from the links and runs the network cleaner afterwards. Thus, some more links may be restricted to keep the network consistent.
+	 * That means, each link can be reached from each other link.
+	 * @param network the network
+	 * @param modesToRemoveByLinkId map of modes that should be removed from the links
+	 */
+	public static void restrictModesAndCleanNetwork(Network network, Map<Id<Link>, Set<String>> modesToRemoveByLinkId) {
```

Comment:
Using a lambda Function<Id<Link>, Set<String> here is probably a lot more flexible.

Reply:
Yes, good idea!